crates/services/tx_status_manager/src/manager.rs

```diff
@@ -508,65 +477,61 @@ mod tests {
             // some old statuses
-            tx_status_manager.status_update(tx4_id, STATUS_1);
-            tx_status_manager.status_update(tx5_id, STATUS_1);
+            tx_status_manager.status_update(tx4_id, status_1());
+            tx_status_manager.status_update(tx5_id, status_1());
             tx_status_manager.status_update(tx1_id, status_2.clone());
             tx_status_manager.status_update(tx2_id, status_2.clone());
 
             // Move beyond TTL
             std::thread::sleep(TTL);
 
             // Trigger the pruning
-            tx_status_manager.status_update(tx6_id, STATUS_1);
+            tx_status_manager.status_update(tx6_id, status_1());
 
             // All but the last one should be pruned.
             assert_absence(
                 &tx_status_manager,
                 vec![tx1_id, tx2_id, tx3_id, tx4_id, tx5_id],
             );
-            assert_presence_with_status(&tx_status_manager, vec![(tx6_id, STATUS_1)]);
+            assert_presence_with_status(&tx_status_manager, vec![(tx6_id, status_1())]);
         }
 
         #[test]
-        fn removes_empty_map_from_cache() {
-            let status_2: TransactionStatus = TransactionStatus::SqueezedOut {
-                reason: "fishy tx".to_string(),
-            };
+        fn status_preserved_in_cache_when_first_status_expires() {
+            let status_2: TransactionStatus =
+                TransactionStatus::squeezed_out("fishy tx".to_string());
 
             let tx_status_change = TxStatusChange::new(100, Duration::from_secs(360));
-            let tx_status_manager = TxStatusManager::new(tx_status_change, TTL);
+            let mut tx_status_manager = TxStatusManager::new(tx_status_change, TTL);
 
             let tx1_id = [1u8; 32].into();
 
             // Register tx1 and remember it's timestamp
-            tx_status_manager.status_update(tx1_id, STATUS_1);
-            let timestamp = {
-                let data = tx_status_manager.inner_data();
-                data.timestamps.keys().copied().next().unwrap()
-            };
+            tx_status_manager.status_update(tx1_id, status_1());
 
             // Sleep for less than a TTL
             std::thread::sleep(Duration::from_secs(2));
 
+            // Given
             // Update tx1 status, the timestamp cache should get updated.
             tx_status_manager.status_update(tx1_id, status_2);
 
-            // Check that there is no stray cache entry for the original timestamp.
-            {
-                let data = tx_status_manager.inner_data();
-                let empty_timestamp_cache_exists =
-                    data.timestamps.contains_key(&timestamp);
-                assert!(!empty_timestamp_cache_exists);
-            }
+            // When
+            // Sleep for TTL, it should expire the first status
+            std::thread::sleep(TTL);
+
+            // Then
+            let status = tx_status_manager.status(&tx1_id);
+            assert!(status.is_some());
```

Comment:
I think this should say `is_none()` (if I got your intention right).
But in such case the test will fail, because we need to call `update_status()` to trigger the pruning, but here we call `sleep` and the directly ask for status.

Reply:
Added more comments amd thank you for spotting not calling the prune logic, added it part by simulating insertion of status for another transaciton=)